feat: pass required context for instructor emails

DmytroAlipov opened the pull request:
## Description

Pass required context to bulk enrollment emails:
- logo_url
- homepage_url
- dashboard_url

Add additional context for enrollment emails:
- contact_email
- platform_name

There is no required context for [base_body.html](https://github.com/openedx/edx-platform/blob/45178e0ced90ea43bce2e6dc0ee7b54b40204755/openedx/core/djangoapps/ace_common/templates/ace_common/edx_ace/common/base_body.html#L81)

openedx-webhooks commented:
Thanks for the pull request, @DmytroAlipov!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.



#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@openedx/wg-maintenance-edx-platform`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

DmytroAlipov commented:
@e0d Here the problem is apparently not in the tests themselves

e0d commented:
@DmytroAlipov rerunning seems to have fixed the failure, will also update the branch.

mphilbrick211 commented:
Hi @openedx/teaching-and-learning! Would someone be able to review/merge this for us? Thanks!

mphilbrick211 commented:
> Hi @openedx/teaching-and-learning! Would someone be able to review/merge this for us? Thanks!

Hi @openedx/teaching-and-learning! Just following up on this.

mphilbrick211 commented:
Hi @DmytroAlipov - flagging the new tests that have popped up per the recent update: https://discuss.openedx.org/t/minor-change-to-edx-platform-check-statuses/11131

DmytroAlipov commented:
done

mphilbrick211 commented:
Hi @openedx/teaching-and-learning! Friendly ping on this.

Commit:
feat: pass required context for instructor emails

pass required context to bulk enrollment emails:
  - logo_url
  - homepage_url
  - dashboard_url
add additional context for enrollment emails:
  - contact_email
  - platform_name

mphilbrick211 commented:
Hi @openedx/teaching-and-learning! Following up on this for review.

navinkarkera commented:
@mphilbrick211 @DmytroAlipov I can help review this PR as CC.

mphilbrick211 commented:
Thank you, @navinkarkera!

navinkarkera commented:
@DmytroAlipov Can you add a bit more context in the description. Were these variables missing before? Why don't we use the whole message_context from `get_base_template_context` function instead of getting these variables one by one?

```python
    param_dict.update(message_context)
```

DmytroAlipov commented:
Hi, @navinkarkera 
There is no required [context for](https://github.com/openedx/edx-platform/blob/45178e0ced90ea43bce2e6dc0ee7b54b40204755/openedx/core/djangoapps/ace_common/templates/ace_common/edx_ace/common/base_body.html#L81) `openedx/core/djangoapps/ace_common/templates/ace_common/edx_ace/common/base_body.html`:
- `homepage_url`
- `logo_url`
- `dashboard_url`
- `platform_name`
- `contact_email`

There is no point in transmitting the entire `message_context` since we need a limited number of parameters.

navinkarkera commented:
@DmytroAlipov Apologies for delay in response. `contact_email` doesn't seem to be used anywhere in `base_body`, are we using it in some other template?

DmytroAlipov commented:
Hi @navinkarkera 

Yes, this is not used. I may have transferred this from our developments and we used it.

If this parameter interferes too much, I can remove it. But if it’s not critical, then let it remain.

DmytroAlipov commented:
@navinkarkera Friendly pign.

navinkarkera commented:
@DmytroAlipov Thanks for the reminder. 

All the [examples of base template context](https://github.com/search?q=repo%3Aopenedx%2Fedx-platform%20get_base_template_context&type=code) seems to use the whole context instead of picking some values from it as the base template infact makes use of these variables. For example: [social_media_urls](https://github.com/openedx/edx-platform/blob/9118c9ab9de8cab8998ab1264979843b01c9090e/themes/red-theme/lms/templates/ace_common/edx_ace/common/base_body.html#L96) etc.

I am not able to understand your reasoning behind selecting some of the variables here.

mphilbrick211 commented:
Hi @DmytroAlipov @navinkarkera - is this still in-progress?

mphilbrick211 commented:
Hi @DmytroAlipov! Are you still pursuing this pull request, or can it be closed?
